include/fatal.h

```diff
@@ -89,26 +89,38 @@ void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *esf);
 void z_fatal_error(unsigned int reason, const z_arch_esf_t *esf);
 
 /**
- * Print messages related to an exception
+ * @brief Print messages related to an exception
  *
- * This ensures the following:
- *  - The log system will enter panic mode if it is not already
- *  - Messages will be sent to printk() or the log subsystem if it is enabled
+ * Messages will be sent to the log subsystem if it is enabled else they will
+ * be sent to printk() (if enabled).
  *
- * Log subsystem filtering is disabled.
  * To conform with log subsystem semantics, newlines are automatically
  * appended, invoke this once per line.
  *
- * @param fmt Format string
- * @param ... Optional list of format arguments
+ * @param ... String with optional list of format arguments
+ */
+#define z_fatal_print(...) \
```

Comment:
You are defining z_fatal_print() in terms of CONFIG_LOG, yet you aren't including <logging/log.h>.